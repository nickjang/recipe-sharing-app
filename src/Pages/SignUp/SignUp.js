import React, { Component } from 'react';
import SignUpForm from '../../Components/SignUpForm/SignUpForm';
import './SignUp.css';

class SignUp extends Component {  
  handleRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {    
    return (
      <SignUpForm onSuccess={this.handleRegistrationSuccess} />
    );
  }
}

SignUp.defaultProps = {
  history: {
    push: () => { }
  }
}

export default SignUp;