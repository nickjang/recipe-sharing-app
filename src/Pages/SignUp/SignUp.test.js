import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUp from './SignUp';

describe(`SignUp component`, () => {
  it('renders SignUp by default', () => {
    const wrapper = shallow(<SignUp />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})