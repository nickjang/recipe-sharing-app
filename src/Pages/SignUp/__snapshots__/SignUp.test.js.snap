// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders SignUp by default 1`] = `
<withRouter(SignUpForm)
  onSuccess={[Function]}
/>
`;
